tests/execution/test-union-fieldtype.md

```diff
@@ -0,0 +1,192 @@
+# Test union type resolve
+
+```graphql @config
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
+  query: Query
+}
+
+union FooBar = Bar | Foo
+
+type Bar {
+  bar: String!
+}
+
+type Foo {
+  foo: String!
+}
+
+type Nested {
+  bar: FooBar @discriminate(field: "type")
+  foo: FooBar @discriminate(field: "type")
```

Comment:
that differs from the suggestion on the issue. On fields you need to specify the same directive for every usage of the union instead of adding directive directly on the union.

Can we implement this on union level instead? that will be less error prone

Reply:
Field is more flexible.